src/components/Appointments/AppointmentDetails.vue

```diff
@@ -125,12 +126,15 @@ export default {
 	},
 	methods: {
 		save() {
+			this.isLoading = true
 			this.$emit('save', {
 				slot: this.timeSlot,
 				description: this.description,
 				email: this.email,
 				displayName: this.displayName,
 				timeZone: this.timeZone,
+			}).then(() => {
+				this.isLoading = false
 			})
 		},
```

Comment:
**suggestion (non-blocking):** Another form is
```suggestion
		async save() {
			this.isLoading = true
			await this.$emit('save', {
				slot: this.timeSlot,
				description: this.description,
				email: this.email,
				displayName: this.displayName,
				timeZone: this.timeZone,
			})
			this.isLoading = false
		},
```

Reply:
Nice, that looks simpler, more DRY. Happy to to adjust if you think that's best!

Reply:
yes please

Reply:
Updated, thanks! :)